### ISSUE TITLE
The improvement for xmpp_stanza_add_child.

### ISSUE THREAD

**Issue by xbr-hw** [strophe/libstrophe#169]
https://github.com/strophe/libstrophe/blob/09229e2cd5e0653d668c0c9b4a5f7f3a3799ada5/src/stanza.c#L692
When the size of stanza is big, the procedure time of get the tail node will be too long. Therefore, we can store then address of the tail node everytime we invoke xmpp_stanza_add_child. Next time we can get the tail node directly. Like this:

int xmpp_stanza_add_child(xmpp_stanza_t *stanza, xmpp_stanza_t *child)
{
    xmpp_stanza_t *s;

    /* get a reference to the child */
    xmpp_stanza_clone(child);

    child->parent = stanza;

    if (!stanza->children)
        stanza->children = child;
    else {
        s = stanza->children_tail;
        s->next = child;
        child->prev = s;
    }
    
    stanza->children_tail = child;
    return XMPP_EOK;
}

**Comment by pasis** [comment]
Good point! I would prefer to implement general-purpose double-linked list and use it everywhere in libstrophe. This will solve the issue. Additionally to performance improvement, this will reduce copy-paste and hide all complexity in a single module.
